### PR TITLE
feat: #351 Add types of buttons, added tooltips to headers

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 1]
```diff
@@ -473,6 +464,45 @@ export const WayPage = (props: WayPageProps) => {
...
+          />
+          <EditableTextarea
+            text={way.goal.description}
+            onChangeFinish={(description) => changeGoalDescription(way.goal, description)}
+            rows={10}
+            isEditable={isOwner}
+            className={styles.goalDescription}
+          />
+        </div>
+        <div className={styles.width}>
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:481]: great naming

File: mw-webapp/src/logic/wayPage/WayStatistic.module.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+@import "src/styles/variables";
+
+.accordion {
+  min-width: 360px;
+  max-width: 600px;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayStatistic.module.scss:5]: variables?

File: mw-webapp/src/logic/wayPage/WayStatistic.tsx [hunk 3]
```diff
@@ -51,55 +60,130 @@ export const WayStatistic = (props: WayStatisticProps) => {
 
   const averageTimeForJob = Math.round(totalWayTime / allJobsAmount.length);
 
-  return (
-    <div>
-      <p>
-        Total days on a way:
-        {" "}
+  const lastWeekDayReports = props.dayReports.filter((dayReport) => {
+    return lastWeekDates.includes(DateUtils.getShortISODateValue(dayReport.createdAt));
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayStatistic.tsx:64]: lets use more performant algoritms

File: mw-webapp/src/logic/wayPage/WayStatistic.tsx [hunk 4]
```diff
@@ -51,55 +60,130 @@ export const WayStatistic = (props: WayStatisticProps) => {
...
+  const lastWeekJobsAmount = lastWeekDayReports.flatMap(report => report.jobsDone);
+
+  const lastCalendarWeekTotalTime = lastWeekJobsAmount.reduce((totalTime, jobDone) => totalTime + jobDone.time, 0);
+
+  const lastCalendarWeekAverageWorkingTime = Math.round(lastCalendarWeekTotalTime / AMOUNT_DAYS_IN_WEEK);
+
+  const lastCalendarWeekAverageJobTime = Math.round(lastCalendarWeekTotalTime / lastWeekJobsAmount.length);
+
+  const lastTwoWeekDayReports = props.dayReports.filter((dayReport) => {
+    return lastTwoWeekDates.includes(DateUtils.getShortISODateValue(dayReport.createdAt));
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayStatistic.tsx:76]: use more performant algoritms

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.module.scss [hunk 5]
```diff
@@ -56,5 +76,11 @@ $listLeftMargin: 20px;
 }
 
 .tooltip {
-  cursor: pointer;
+  width: 120px;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.module.scss:79]: wrong approach

File: mw-webapp/src/logic/waysTable/OwnWaysTable.tsx [hunk 6]
```diff
@@ -7,37 +9,38 @@ import {WAYS_OWNER, waysColumns} from "src/logic/waysTable/waysColumns";
...
+export const OwnWaysTable = (props: OwnWaysTableProps) => {
   const [ownWays, setOwnWays] = useState<WayPreview[]>([]);
+  const userPreviewUuid = props.uuid;
 
   /**
    * Load User own ways
    */
   const loadOwnWays = async () => {
-    const data = await WayPreviewDAL.getUserWaysPreview(props.uuid, "Own");
+    const data = await WayPreviewDAL.getUserWaysPreview(userPreviewUuid, "Own");
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/OwnWaysTable.tsx:37]: add TODO

File: mw-webapp/src/logic/waysTable/columns.module.scss [hunk 7]
```diff
@@ -15,18 +17,7 @@
...
-  word-wrap: break-word;
-  @media (max-width: 800px) {
-    -webkit-line-clamp: 5;
-  }
-}
-
-.tooltip {
-  cursor: pointer;
+  display: flex;
+  min-width: 80px;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/columns.module.scss:21]: variab;e

File: mw-webapp/src/utils/DateUtils.ts [hunk 8]
```diff
@@ -15,4 +15,19 @@ export class DateUtils {
       .substring(START_OF_ISO_SUBSTRING_RANGE, END_OF_ISO_SUBSTRING_RANGE);
   }
 
+  /**
+   * Get array with last {@link amount} dates
+   */
+  public static getLastDates(amount: number): string[] {
```
Comment (tritonJS826) [inline mw-webapp/src/utils/DateUtils.ts:21]: do we still need it?)

Comment (Ekaterina1994) [reply mw-webapp/src/utils/DateUtils.ts]: Deleted

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.module.scss (hunk 5 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.module.scss]: deleted

File: mw-webapp/src/logic/wayPage/WayStatistic.tsx (hunk 4 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/WayStatistic.tsx]: Done

File: mw-webapp/src/logic/wayPage/WayStatistic.tsx (hunk 3 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/WayStatistic.tsx]: Done

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 9]
```diff
@@ -263,17 +244,18 @@ export const Columns = (props: ColumnsProps) => {
...
-                  </Tooltip>
-                }
-              </div>
+              {isOwner &&
+              <Tooltip
+                content="Add job"
+                position={PositionTooltip.RIGHT}
+              >
+                <Button
+                  value="➕"
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:253]: Should it be symbols plus?

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**